src/System.Private.ServiceModel/tests/Scenarios/Client/TypedClient/TypedProxyDuplexTests.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using TestTypes;
+using Xunit;
+
+public static class TypedProxyDuplexTests
+{
+    // ServiceContract typed proxy tests create a ChannelFactory using a provided [ServiceContract] Interface which...
+    //       returns a generated proxy based on that Interface.
+    // ChannelShape typed proxy tests create a ChannelFactory using a WCF understood channel shape which...
+    //       returns a generated proxy based on the channel shape used, such as...
+    //              IRequestChannel (for a request-reply message exchange pattern)
+    //              IDuplexChannel (for a two-way duplex message exchange pattern)
+
+    [Fact]
+    [ActiveIssue(138)]
+    [OuterLoop]
+    public static void ServiceContract_TypedProxy_AsyncTask_CallbackReturn()
+    {
+        DuplexChannelFactory<IWcfDuplexTaskReturnService> factory = null;
+        Guid guid = Guid.NewGuid();
+
+        NetTcpBinding binding = new NetTcpBinding();
+        binding.Security.Mode = SecurityMode.None;
+
+        DuplexChannelServiceCallback callbackService = new DuplexChannelServiceCallback();
+        InstanceContext context = new InstanceContext(callbackService);
+
+        using (factory = new DuplexChannelFactory<IWcfDuplexTaskReturnService>(context, binding, new EndpointAddress(Endpoints.Tcp_NoSecurity_Callback_Address)))
+        {
+            IWcfDuplexTaskReturnService serviceProxy = factory.CreateChannel();
+
+            Task<Guid> task = serviceProxy.Ping(guid);
+
+            Guid returnedGuid = task.Result;
+
+            if (guid != returnedGuid)
+            {
+                Assert.True(false, String.Format("The sent GUID does not match the returned GUID. Sent: {0} Received: {1}", guid, returnedGuid));
```

Comment:
Assert.Equal looks good, thanks.

"guid" is first sent to the service operation, then passed to the callback server side, the server callback sends it to the callback on the client which just returns it back to the server callback, the service operation then returns it to the client as the return value of the initial call to the service operation as "returnedGuid", at least that's the intention, unless I wired something together wrong.